From 3f42f727a0699f7274a99ea39def7f9b4c3b0c1e Mon Sep 17 00:00:00 2001
From: Jussi Kivilinna <jussi.kivilinna@iki.fi>
Date: Sun, 31 Jan 2021 16:59:37 +0200
Subject: [PATCH 6/6] Make sure the grcy_get_config string is always
 null-terminated.

* src/global.c (_gcry_get_config): Append null-terminator to output.
--

Config string was not being explicitly null-terminated which resulted
garbage output from tests/version with ASAN enabled builds.

Signed-off-by: Jussi Kivilinna <jussi.kivilinna@iki.fi>
---
 src/global.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/global.c b/src/global.c
index cd3d9d90..7d634095 100644
--- a/src/global.c
+++ b/src/global.c
@@ -418,6 +418,10 @@ _gcry_get_config (int mode, const char *what)
     return NULL;
 
   print_config (what, fp);
+
+  /* Make sure the output is null terminated. */
+  gpgrt_fwrite ("", 1, 1, fp);
+
   if (gpgrt_ferror (fp))
     {
       save_errno = errno;
-- 
2.30.0

